src/hashes/blake2b.c

```diff
@@ -342,7 +342,7 @@ int blake2b_process(hash_state *md, const unsigned char *in, unsigned long inlen
       unsigned long fill = BLAKE2B_BLOCKBYTES - left;
       if (inlen > fill) {
          md->blake2b.curlen = 0;
-         XMEMCPY(md->blake2b.buf + left, in, fill); /* Fill buffer */
+         XMEMCPY(md->blake2b.buf + (left % sizeof(md->blake2b.buf)), in, fill); /* Fill buffer */
```

Comment:
I decided to use `sizeof(md->blake2b.buf)` so it's clear that we're truncating `left` by the size of the buffer, without having to lookup how large `md->blake2b.buf` really is